flask/templating.py

```diff
@@ -121,8 +121,24 @@ def render_template(template_name_or_list, **context):
     """
     ctx = _app_ctx_stack.top
     ctx.app.update_template_context(context)
-    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
-                   context, ctx.app)
+
+    template = None
+
+    if request.blueprint is not None and \
+            isinstance(template_name_or_list, string_types):
+        bp = current_app.blueprints[request.blueprint]
```

Comment:
Also `_request_ctx_stack` contains the request, therefore no need to import `current_app` and `request`.